include/services/daal_atomic_int.h

```diff
@@ -51,39 +58,66 @@ class DAAL_EXPORT Atomic
      * Returns an increment of atomic object
      * \return An increment of atomic object
      */
-    dataType inc();
+    dataType inc()
+    {
+        tbb::atomic<dataType> * atomicPtr = (tbb::atomic<dataType> *)(this->_ptr);
```

Comment:
1) Better to use static_cast instead of C-cast
2) check for null  pointer is required

The same for other places

Reply:
existing constructors ensure no null pointer is possible